colossalai/shardformer/modeling/bert.py

```diff
@@ -177,6 +180,14 @@ def bert_model_forward(
         start_idx, end_idx = stage_index[0], stage_index[1]
         # layer_outputs
         layer_outputs = hidden_states if hidden_states is not None else None
+
+        # split the input tensor along sequence dimension
+        # [batch_size, seq_len, hidden_size] -> [batch_size, seq_len/TP_size, hidden_size]
```

Comment:
What if seq len is not dividable by tp size?

Reply:
Inside the _split function, there is an assert for dim_size // world_size.

Reply:
> What if seq len is not dividable by tp size?

Inside the _split function, there is an assert for dim_size // world_size.